core/credential/credential.go

```diff
@@ -0,0 +1,44 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package credential
+
+import "context"
+
+type Credentials struct {
+	Host     string
+	Username string
+	Secret   string
+}
+
+// Manager is Credential manager plugin core method
+type Manager interface {
+	Store(ctx context.Context, cred Credentials) error
+	Get(ctx context.Context, host string) (Credentials, error)
```

Comment:
We should have a Get interface which can support this auth request interface https://github.com/containerd/containerd/blob/main/api/types/transfer/registry.proto#L45

Reply:
I implemented the `CredentialHelper` interface to solve the callback problem of obtaining `Credential`. What do you think of this implementation?

Reply:
Replace old `Get` method.